pulumi/grapl/__main__.py

```diff
@@ -239,115 +226,54 @@ def _get_integration_test_job_vars(
             )
 
     else:
-        nomad_cluster = NomadCluster(
-            "nomad-cluster",
-            network=network,
-        )
-
-        # No Fargate or Elasticache in Local Grapl
         cache = Cache("main-cache", network=network)
+        pulumi_config = pulumi.Config()
 
-        sysmon_generator = SysmonGenerator(
-            input_emitter=sysmon_log_emitter,
-            output_emitter=unid_subgraphs_generated_emitter,
-            network=network,
-            cache=cache,
-            forwarder=forwarder,
-        )
-
-        osquery_generator = OSQueryGenerator(
-            input_emitter=osquery_log_emitter,
-            output_emitter=unid_subgraphs_generated_emitter,
-            network=network,
-            cache=cache,
-            forwarder=forwarder,
-        )
-
-        node_identifier = NodeIdentifier(
-            input_emitter=unid_subgraphs_generated_emitter,
-            output_emitter=subgraphs_generated_emitter,
-            db=dynamodb_tables,
-            network=network,
-            cache=cache,
-            forwarder=forwarder,
-        )
-
-        graph_merger = GraphMerger(
-            input_emitter=subgraphs_generated_emitter,
-            output_emitter=subgraphs_merged_emitter,
-            dgraph_cluster=dgraph_cluster,
-            db=dynamodb_tables,
-            network=network,
-            cache=cache,
-            forwarder=forwarder,
-        )
-
-        analyzer_dispatcher = AnalyzerDispatcher(
-            input_emitter=subgraphs_merged_emitter,
-            output_emitter=dispatched_analyzer_emitter,
-            analyzers_bucket=analyzers_bucket,
-            network=network,
-            cache=cache,
-            forwarder=forwarder,
-        )
-
-        analyzer_executor = AnalyzerExecutor(
-            input_emitter=dispatched_analyzer_emitter,
-            output_emitter=analyzer_matched_emitter,
-            dgraph_cluster=dgraph_cluster,
-            analyzers_bucket=analyzers_bucket,
-            model_plugins_bucket=model_plugins_bucket,
-            network=network,
-            cache=cache,
-            forwarder=forwarder,
+        grapl_core_job_aws_vars = pulumi.Output.all(**nomad_inputs,).apply(
+            lambda inputs: {
+                "container_registry": "docker.cloudsmith.io/",
+                "container_repo": "raw/",
+                "grapl_test_user_name": config.GRAPL_TEST_USER_NAME,
+                "_redis_endpoint": cache.endpoint,
+                # TODO: consider replacing with the previous per-service `configurable_envvars`
+                "rust_log": "DEBUG",
+                # Build Tags. We use per service tags so we can update services independently
+                "analyzer_dispatcher_tag": pulumi_config.require_object("artifacts")[
+                    "analyzer-dispatcher"
+                ],
```

Comment:
I'd just extract the artifacts object somewhere above this and then just pull out the thing I needed, like `artifacts["analyzer-dispatcher"]`

Reply:
Incidentally, this is where you'd stick the logic for composing the container image names: `analyzer_dispatcher_image: image_name("analyzer-dispatcher", artifacts, registry, repo)`

(though we can come up with a nicer interface.)